### DISCUSSION TITLE
How do I resolve a UnicodeDecodeError when using Sling?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17966] [Q&A] [answered]
Hey there, i'm trying to use dagster-embeded-elt, but whenever I try to run the asset, I'm getting this error

```
source = SlingSourceConnection(
    type="sqlserver",
    host="host",
    port=1433,
    database="database",
    user="user",
    password=quote_plus('password'),
)

target = SlingTargetConnection(
    type="s3",
    bucket=os.getenv('S3_ENV'),
    access_key_id=os.getenv("S3_ENV"),
    secret_access_key=os.getenv("S3_ENV")
)
sling = SlingResource(source_connection=source, target_connection=target)

asset_def = build_sling_asset(
    asset_spec=AssetSpec("elt_dagster"),
    source_stream="TABLE",
    target_object="<s3://bucket/dagster-elt/table/>",
    target_options={'file_max_rows': 100000, 'format': "parquet"},
    mode=SlingMode.FULL_REFRESH
)
```
![image](https://github.com/dagster-io/dagster/assets/1045990/e57dc262-163c-4451-bffa-8c29f867346c)
![image](https://github.com/dagster-io/dagster/assets/1045990/44cf65a0-f01c-46ba-9fa3-b1bef605167e)

Am I doing anything wrong?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699901545150469?thread_ts=1699901545.150469&cid=C01U954MEER).*

**Comment by PedramNavid** [comment]
Hi! 

Could you provide a few more details? This looks like an encoding issue.

What platform are you running on, e.g. Mac, Windows, Linux.
What version of Python are you using?
What language are you set to on your machine?

**Reply by IsmaelRDeMelo** [reply]
Windows
Python 3.9
pt-br

**Reply by PedramNavid** [reply, marked as answer]
Thanks @IsmaelRDeMelo! Should be fixed by this: https://github.com/dagster-io/dagster/pull/18055
